python/GafferSceneTest/EvaluateLightLinksTest.py

```diff
@@ -0,0 +1,119 @@
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import unittest
+
+import IECore
+
+import Gaffer
+import GafferScene
+import GafferSceneTest
+import GafferArnold
+
+class EvaluateLightLinksTest( GafferSceneTest.SceneTestCase ) :
+
+	def test( self ) :
+
+		sphere = GafferScene.Sphere( "Sphere" )
+
+		attributes = GafferScene.StandardAttributes()
+		attributes["attributes"]["linkedLights"]["enabled"].setValue( True )
+		attributes["in"].setInput( sphere["out"] )
+
+		mainGroup = GafferScene.Group( "MainGroup" )
+		mainGroup["in"].addChild( GafferScene.ScenePlug( "in1" ) )
+		mainGroup["in"].addChild( GafferScene.ScenePlug( "in2" ) )
+
+		light1 = GafferArnold.ArnoldLight()
+		light1.loadShader( "point_light" )
+
+		light2 = GafferArnold.ArnoldLight()
+		light2.loadShader( "point_light" )
+
+		lightGroup = GafferScene.Group( "LightGroup" )
+		lightGroup["in"].addChild( GafferScene.ScenePlug( "in1" ) )
+		lightGroup["in"].addChild( GafferScene.ScenePlug( "in2" ) )
+		lightGroup["in"]["in1"].setInput( light1["out"] )
+		lightGroup["in"]["in2"].setInput( light2["out"] )
+
+		mainGroup["in"]["in1"].setInput( attributes["out"] )
+		mainGroup["in"]["in2"].setInput( lightGroup["out"] )
+
+		lightSet = GafferScene.Set( "lightSet" )
+		lightSet["in"].setInput( mainGroup["out"] )
+		lightSet["name"].setValue( 'lightSet' )
+		lightSet["paths"].setValue( IECore.StringVectorData( [ '/group/group/light', '/group/group/light1' ] ) )
+
+		evalNode = GafferScene.EvaluateLightLinks()
+		evalNode["in"].setInput( lightSet["out"] )
+
+		# Test a single light
+		attributes["attributes"]["linkedLights"]["value"].setValue( "/group/group/light" )
+
+		self.assertEqual(
+			evalNode["out"].attributes( "/group/sphere" )[ "gaffer:linkedLights" ],
+			IECore.StringVectorData( [ "/group/group/light" ] ) )
+
+		# Test a simple list of lights
+		attributes["attributes"]["linkedLights"]["value"].setValue( "/group/group/light /group/group/light1" )
+
+		self.assertEqual(
+			evalNode["out"].attributes( "/group/sphere" )[ "gaffer:linkedLights" ],
+			IECore.StringVectorData( [ "/group/group/light", "/group/group/light1" ] ) )
+
+		# Make sure only lights come through
+		attributes["attributes"]["linkedLights"]["value"].setValue( "/group/group/light /group/group/light1 /group/sphere" )
+
+		self.assertEqual(
+			evalNode["out"].attributes( "/group/sphere" )[ "gaffer:linkedLights" ],
+			IECore.StringVectorData( [ "/group/group/light", "/group/group/light1" ] ) )
+
+		# Make sure changing sets updates the result of the expression evaluation
```

Comment:
I was wondering if you'd like these to be separate test methods? Do we have a convention for setting up a test script and using it in mulitple tests? Creating the entire graph over and over again for every test seems a little ugly.